### PR TITLE
GenericConstantRankTwoTensor.C instantiating the templated classes.

### PR DESCRIPTION
**Pull request by lekah**
Those lines missing can create problems when inheriting from GenericConstantRankTwoTensor in an application (seems to be compiler dependent). There is no issue so far referencing this problem as this is very minor.

## Reason
Inheriting from GenericConstantRankTwoTensor creates problems for some compilers. Many other templated classes are also being instantiated, so I figured there is no reason for this not to be the case here.

### CODE REVIEW

**Comment by GiudGiud** [comment]
@lekah we need an issue to be named in the commit, even for very minor items. If you think it s too trivial, you can link it to the one that was created for this object, or that  was created for this way of templating with AD. But we mind 0% if you open a new one

File: framework/src/materials/GenericConstantRankTwoTensor.C [hunk 1]
```diff
@@ -50,3 +50,6 @@ GenericConstantRankTwoTensorTempl<is_ad>::computeQpProperties()
 {
   _prop[_qp] = _tensor;
 }
+
+template class GenericConstantRankTwoTensorTempl<false>;
+template class GenericConstantRankTwoTensorTempl<true>;
```
Comment (GiudGiud) [inline framework/src/materials/GenericConstantRankTwoTensor.C:55]: also this will need a new line at the end of file

**Comment by lekah** [comment]
@GiudGiud  I've updated the PR and link to a new issue https://github.com/idaholab/moose/issues/22619

**Review by GiudGiud — APPROVED** [review]
thanks for the patch!

**Pull request merged**